src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Storage storage;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy, Storage storage) {
+        this.operationStrategy = operationStrategy;
+        this.storage = storage;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
```

Comment:
Avoid using null checks for the strategy pattern. Instead, ensure that your `OperationStrategy` always returns a valid `OperationHandler`. This can be achieved by initializing the strategy map with all necessary operations at the start of the application.